target/linux/ath79/dts/qca9563_dlink_dir-869-a1.dts

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca956x.dtsi"
+
+/ {
+	model = "D-Link DIR-869 A1";
+	compatible = "dlink,dir-869-a1", "qca,qca9563";
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+
+		led_power: power {
+			label = "dir-869-a1:orange:power";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		internet {
+			label = "dir-869-a1:white:internet";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		wps {
+			linux,code = <KEY_WPS_BUTTON>;
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+
+		reset {
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
```

Comment:
Are those really swapped compared to the dir-859?

Reply:
I hope not to sound too much like a noob. This is my first try into coding with github at all. So there are probably a dozen beginners mistakes.

> Please add specifications, flashing instructions and an overview of MAC addresses used in the vendor firmware to the commit message, similar to previous submissions for similar devices.

Can you please point me in the right direction for this? I guess you also mean the hardware specs, instead of just MAC addresses. I did see some commits with digits added for certain interfaces. I guess that is what you mean, what digit is used for which interface. Am I right about this?

I did not use the vendor firmware for creating this patch. I started by interpreting the ar71xx source, putting that next to the dir-859 source and basically used the dir-859 as a template, where I added dir-869 values where needed. 

> Are those really swapped compared to the dir-859?

Yes they are. I took my gpio values from this file. https://git.openwrt.org/?p=openwrt/openwrt.git;a=blob;f=target/linux/ar71xx/files/arch/mips/ath79/mach-dir-869-a1.c;h=b0894e37cce9135d37ebf12847d8685d048f4ae1;hb=c4c72a9a42e07bed8f9e9eb214c0debfb5d7ab90



> Please create a shared DTSI with the dir-859-a1, similar to how it's done for the dir-842-cs ...

I have to look into that. But I guess that will be manageable.

Reply:
Commit message:

Just look at what people did before you (preferably in recent patches), e.g.
https://github.com/openwrt/openwrt/commit/68ac3f2cddab8422d7de0ce1a78d23edf29012e7

> I did not use the vendor firmware for creating this patch. I started by interpreting the ar71xx source, putting that next to the dir-859 source and basically used the dir-859 as a template, where I added dir-869 values where needed.

Then we will have to live with that. In the past, many people just invented their own MAC address assignment scheme, so actually checking with OEM firmware might be helpful to find the true assignment. However, that's optional if there is ar71xx setup already.

Reply:
Mode switch is missing here.